docs/docs/50-user-guide/60-reference-docs/80-webhook-receivers/generic.md

```diff
@@ -4,8 +4,267 @@ sidebar_label: Generic
+3. [By Values in an Index](#by-values-in-an-index)
+
+All methods support both static values and 
+[expression-derived values](#expression-functions).
+
+:::note
+Using more than one of the above results in a criteria set that is the logical
+**AND** of all defined criteria.
 :::
+
+#### By Name
+
+The simplest way to designate a resource is by specifying its `name`.
+
+##### Example
+
+```yaml
+apiVersion: kargo.akuity.io/v1alpha1
+kind: ProjectConfig
+metadata:
+  name: kargo-demo
+  namespace: kargo-demo
+spec:
+  webhookReceivers:
+    - name: my-receiver
+      generic:
+        secretRef:
+          name: wh-secret
+      actions:
+        - actionType: Refresh
+          whenExpression: "request.header('X-Event-Type') == 'push'"
+          targetSelectionCriteria:
+            # Static name designation
+            - kind: Warehouse
+              name: my-warehouse
+            # Expression-derived name designation
+            - kind: Warehouse
+              name: "${{ normalizeGit(request.body.repository.name) }}"
+```
+
+#### By Labels
+
+Use `labelSelector` to designate resources based on labels. It supports both 
+`matchLabels` and `matchExpressions`.
+
+##### Example
+
+```yaml
+apiVersion: kargo.akuity.io/v1alpha1
+kind: ProjectConfig
+metadata:
+  name: kargo-demo
+  namespace: kargo-demo
+spec:
+  webhookReceivers:
+    - name: my-receiver
+      generic:
+        secretRef:
+          name: wh-secret
+      actions:
+        - actionType: Refresh
+          whenExpression: "request.header('X-Event-Type') == 'push'"
+          targetSelectionCriteria:
+            - kind: Warehouse
+              labelSelector:
+                matchLabels:
+                  # Warehouses with the 'environment' label set to 'prod'
+                  environment: prod
+                matchExpressions:
+                  # Warehouses with the 'service' label set to 'ui' OR 'api'
+                  - key: service
+                    operator: In
+                    values: ["ui", "api"]
+```
+
+#### By Values in an Index
+
+Use `indexSelector` to retrieve resources by a cached index.
+
+##### Example
+
+```yaml
+actions:
+  - actionType: Refresh
+    whenExpression: "request.header('X-Event-Type') == 'push'"
+    targetSelectionCriteria:
+      - kind: Warehouse
+        indexSelector:
+          matchIndices:
+            - key: subscribedURLs
+              operator: Equals
+              value: "${{ normalizeGit(request.body.repository.url) }}"
+```
+
+:::note
+`subscribedURLs` is the only available index. It refers to `Warehouse` 
+resources that contain subscriptions for a provided repository URL.
+:::
+
+### Expression functions
+
+The Generic webhook receiver extends
+[built-in expr-lang support](https://expr-lang.org/docs/language-definition) 
+with utilities that can be used to help derive `TargetSelectionCriteria` 
+information from incoming requests.
+
+The following expression functions are available:
+
+#### request.body
```

Comment:
The first thing listed under available functions is not a function. In terms of how to rectify that, I might generalize "### Expression functions" into something like "### Expression Reference" with some introductory texts about functions _and variables_ that are predefined and available for use in this context.